projects/ruby/fuzz_ruby_gems.c

```diff
@@ -0,0 +1,400 @@
+/* Copyright 2022 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "init_ruby_load_paths.h"
+#include <ruby/ruby.h>
+#include <unistd.h>
+
+#define ARRAYSIZE(x) (sizeof(x) / sizeof(x[0]))
+
+enum RubyDataType { RDT_CString };
+
+struct TargetFunction {
+  const char *module_;
+  const char *cls_;
+  const char *name_;
+  VALUE module_obj_;
+  VALUE cls_obj_;
+  ID method_id_;
+  int nargs_;
+  const enum RubyDataType *argTypes_;
+};
+
+struct TargetCall {
+  struct TargetFunction *fcn_;
+  VALUE *args_;
+};
+
+VALUE eval(const char *cmd) {
+  int state = 0;
+  VALUE result = rb_eval_string_protect(cmd, &state);
+  if (state != 0) {
+    rb_set_errinfo(Qnil);
+  }
+
+  return result;
+}
+
+static VALUE call_protected_helper(VALUE rdata) {
+  struct TargetCall *call = (struct TargetCall *)(rdata);
+  struct TargetFunction *fcn = call->fcn_;
+  return rb_funcall2(fcn->obj_, fcn->method_id_, fcn->nargs_, call->args_);
+};
+
+VALUE call_protected(struct TargetCall *call) {
+  int state = 0;
+  VALUE result = rb_protect(call_protected_helper, (VALUE)(call), &state);
+  if (state != 0) {
+    rb_set_errinfo(Qnil);
+  }
+  return result;
+}
+
+VALUE require(const char *module) {
+  char cmd[1024];
+  snprintf(cmd, sizeof(cmd), "require '%s'\n", module);
+  return eval(cmd);
```

Comment:
Should this be `rb_require`?

Reply:
Thanks, I've replaced it with `rb_require`.